src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,67 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = '', health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      const index = Animal.alive.findIndex((animal) => animal === this);
+
+      Animal.alive.splice(index, 1);
```

Comment:
The code is using `findIndex` followed by `splice` to remove the animal from the `Animal.alive` array when its health reaches 0 or less. As per the checklist, avoid nesting loops and consider using `filter` instead for better performance and cleaner code.